components/tikv_util/src/memory.rs

```diff
@@ -47,6 +48,13 @@ pub trait HeapSize {
     fn approximate_heap_size(&self) -> usize {
         0
     }
+
+    fn approximate_mem_size(&self) -> usize
```

Comment:
Why do we need it? If the memory is in stack, we don't need to care it.

Reply:
Because stack memory is memory too, it may also cause OOM 😂

Reply:
But stack memory is just several MB, is it a problem?

Reply:
Oh, I see `tracker` is wrapped in `Box`. Seems we should always consider `mem::size_of::<Self>` for `approximate_heap_size`. Do we really need a separated interface `approximate_mem_size` apart from `approximate_heap_size`

Reply:
> Why do we need it? If the memory is in stack, we don't need to care it.
> Because stack memory is memory too, it may also cause OOM 😂

The term "memory in stack" is bit ambiguous. What I mean is the memory used by a struct itself, and it is not necessary to always stay in thread stack because it may be moved to heap later on.

> But stack memory is just several MBs, is it a problem?

It does matter, consider about `[u8; 1024*1024*1024]`. It has zero heap allocation while itself takes 1GB memory.

> Do we really need a separated interface approximate_mem_size apart from approximate_heap_size

It's a short cut of getting the total memory usage of a struct, much like the idea of provided methods in Iterator trait.